libs/langchain/langchain/tools/ddg_search/tool.py

```diff
@@ -56,14 +56,24 @@ class DuckDuckGoSearchResults(BaseTool):
     api_wrapper: DuckDuckGoSearchAPIWrapper = Field(
         default_factory=DuckDuckGoSearchAPIWrapper
     )
+    backend: str = Field(default="api")
 
     def _run(
         self,
         query: str,
         run_manager: Optional[CallbackManagerForToolRun] = None,
     ) -> str:
         """Use the tool."""
-        return str(self.api_wrapper.results(query, self.num_results))
+        res = self.api_wrapper.results(query, self.num_results, backend=self.backend)
+
+        def dict2str(d: dict):
+            items = []
+            for k, v in d.items():
+                items.append(f"{k}: {v}")
+            return ", ".join(items)
```

Comment:
dict to str:
`str(d)`
or
`json.dums(d)`

Reply:
Yes, had this before, but all the keys and values are in quotation marks. Though that could result in an ugly prompt when using the tool in an agent and might lead to worse results in a LLM?! Also, with this approach the output is more similar to the output of DuckDuckGoSearchRun